fontra2fontir/src/source.rs

```diff
@@ -209,16 +226,114 @@ impl FontData {
 
 #[derive(Debug, Clone, Deserialize)]
 struct FontraAxis {
-    _name: String,
-    _tag: Tag,
+    name: String,
+    tag: Tag,
+    #[serde(default)]
+    hidden: bool,
     #[serde(rename = "minValue")]
-    _min_value: f64,
+    min_value: f64,
     #[serde(rename = "defaultValue")]
-    _default_value: f64,
+    default_value: f64,
     #[serde(rename = "maxValue")]
-    _max_value: f64,
+    max_value: f64,
     #[serde(default)]
-    _mapping: Vec<[f64; 2]>,
+    mapping: Vec<[f64; 2]>,
+}
+
+#[derive(Debug)]
+struct StaticMetadataWork {
+    fontdata_file: PathBuf,
+    glyph_info: Arc<BTreeMap<GlyphName, (PathBuf, Option<u32>)>>,
+}
+
+fn create_static_metadata(fontdata_file: &Path) -> Result<StaticMetadata, WorkError> {
+    debug!("Static metadata for {:#?}", fontdata_file);
+    let font_data = FontData::from_file(fontdata_file)
+        .map_err(|e| WorkError::ParseError(fontdata_file.to_path_buf(), format!("{e}")))?;
+
+    let axes = font_data
+        .axes
+        .iter()
+        .map(|a| {
+            let min = UserCoord::new(a.min_value as f32);
+            let default = UserCoord::new(a.default_value as f32);
+            let max = UserCoord::new(a.max_value as f32);
+
+            if min > default || max < default {
+                return Err(WorkError::InconsistentAxisDefinitions(format!("{a:?}")));
+            }
+
+            let converter = if !a.mapping.is_empty() {
+                let examples: Vec<_> = a
+                    .mapping
+                    .iter()
+                    .map(|[raw_user, raw_design]| {
+                        (
+                            UserCoord::new(*raw_user as f32),
+                            DesignCoord::new(*raw_design as f32),
+                        )
+                    })
+                    .collect();
+                let default_idx = examples
+                    .iter()
+                    .position(|(u, _)| *u == default)
+                    .ok_or_else(|| WorkError::AxisMustMapDefault(a.tag))?;
+                examples
+                    .iter()
+                    .position(|(u, _)| *u == min)
+                    .ok_or_else(|| WorkError::AxisMustMapMin(a.tag))?;
```

Comment:
this might be easier with `Vec::contains`?

Reply:
Examples has (User, Design) and I want to search by only User.